src/library/scala/collection/IterableLike.scala

```
@@ -172,12 +173,25 @@ self =>
    *  @return An iterator producing ${coll}s of size `size`, except the
    *          last will be less than size `size` if the elements don't divide evenly.
    */
-  def grouped(size: Int): Iterator[Repr] =
-    for (xs <- iterator grouped size) yield {
+  def grouped(size: Int): Iterator[Repr] = {
+    val arrayBufferSize = {
+      val itSize = iterator.size
+      itSize / size + itSize % size
```

If size is negative (or zero) this will result in weird stuff.  It's also wrong: try it out with itSize = 100 and size=50.  (You end up with 27 slots but you only need 2.)

You should throw an exception on size <= 0 like `GroupedIterator` does.

---

you're absolutely right, this was very naive... Thanks for the feedback.